### PR TITLE
[Flow] Upgrade to 70

### PR DESCRIPTION
**Pull request by jasonLaster**
Fixes Issue: #5687

### Summary of Changes

* starting with 68 here #6501

### CODE REVIEW

File: src/components/QuickOpenModal.js [hunk 1]
```diff
@@ -323,7 +323,15 @@ export class QuickOpenModal extends Component<Props, State> {
...
   /* eslint-disable react/no-danger */
-  renderHighlight = (candidateString: string, query: string, name: string) => {
+  renderHighlight(
+    candidateString: string | React$Element<"div">,
+    query: string,
+    name: string
+  ): string | React$Element<"div"> {
+    if (typeof candidateString != "string") {
+      return "";
+    }
```
Comment (jasonLaster) [inline src/components/QuickOpenModal.js]: maybe we should only call this if the candidate is a string

File: src/components/QuickOpenModal.js [hunk 2]
```diff
@@ -332,21 +340,26 @@ export class QuickOpenModal extends Component<Props, State> {
...
     return <div dangerouslySetInnerHTML={{ __html: html }} />;
-  };
+  }
+
+  updateResult(result: QuickOpenResult, newQuery: string): QuickOpenResult {
+    return {
+      ...result,
+      title: this.renderHighlight(result.title, basename(newQuery), "title")
+    };
+  }
```
Comment (jasonLaster) [inline src/components/QuickOpenModal.js]: i moved this to its own function to simplify the typing but i think we can move it back...

File: src/components/SecondaryPanes/Frames/Frame.js [hunk 3]
```diff
@@ -87,7 +87,7 @@ export default class FrameComponent extends Component<FrameComponentProps> {
     frame: Frame,
     selectedFrame: Frame
   ) {
-    if (e.nativeEvent.which == 3) {
+    if (e.which == 3) {
```
Comment (jasonLaster) [inline src/components/SecondaryPanes/Frames/Frame.js]: i couldnt find nativeEvent in the types... lets see if this still works

File: src/utils/quick-open.js [hunk 4]
```diff
@@ -70,7 +71,7 @@ export function formatSourcesForList(source: RelativeSource, tabs: TabList) {
 export type QuickOpenResult = {|
   id: string,
   value: string,
-  title: string,
+  title: string | React$Element<"div">,
```
Comment (jasonLaster) [inline src/utils/quick-open.js]: when we use fuzzaldrin to fuzzy highlight we return a React Element

File: src/workers/parser/getSymbols.js [hunk 5]
```diff
@@ -145,7 +148,7 @@ function extractSymbol(path: SimplePath, symbols) {
   if (t.isClassDeclaration(path)) {
     symbols.classes.push({
       name: path.node.id.name,
-      parent: path.node.superClass,
+      parent: { name: path.node.superClass },
```
Comment (jasonLaster) [inline src/workers/parser/getSymbols.js]: we had a bug in inComponent that needed this

File: src/workers/parser/getSymbols.js [hunk 6]
```diff
@@ -291,7 +294,8 @@ function extractSymbols(sourceId): SymbolDeclarations {
     imports: [],
     literals: [],
     hasJsx: false,
-    hasTypes: false
+    hasTypes: false,
+    loading: false
```
Comment (jasonLaster) [inline src/workers/parser/getSymbols.js]: we can no longer check if a property exists w/ flow... with that said, we might want to try another option here

Comment (wldcordeiro) [reply src/workers/parser/getSymbols.js]: 👍 interesting note, just saw that in 0.68 release notes.

File: src/utils/quick-open.js (hunk 4 above)
Comment (wldcordeiro) [reply src/utils/quick-open.js]: 👍

File: src/workers/parser/getSymbols.js (hunk 5 above)
Comment (wldcordeiro) [reply src/workers/parser/getSymbols.js]: Hmm, this seems to break preview however as the snapshots are not generating properly.

Comment (wldcordeiro) [reply src/workers/parser/getSymbols.js]: I'm going to revert this change, if there's a bug we should open an issue for it. 😄 

screen shot 2018-06-18 at 9 19 45 am

**Pull request merged**